types.go

```diff
@@ -0,0 +1,114 @@
+package flixkit
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type Network struct {
+	Address        string `json:"address"`
+	FqAddress      string `json:"fq_address"`
+	Contract       string `json:"contract"`
+	Pin            string `json:"pin"`
+	PinBlockHeight uint64 `json:"pin_block_height"`
+}
+
+type Argument struct {
+	Index    int      `json:"index"`
+	Type     string   `json:"type"`
+	Messages Messages `json:"messages"`
+	Balance  string   `json:"balance"`
+}
+
+type Title struct {
+	I18N map[string]string `json:"i18n"`
+}
+
+type Description struct {
+	I18N map[string]string `json:"i18n"`
+}
+
+type Messages struct {
+	Title       *Title       `json:"title,omitempty"`
+	Description *Description `json:"description,omitempty"`
+}
+
+type Dependencies map[string]Contracts
+type Contracts map[string]Networks
+type Networks map[string]Network
+type Arguments map[string]Argument
+
+type Data struct {
+	Type         string       `json:"type"`
+	Interface    string       `json:"interface"`
+	Messages     Messages     `json:"messages"`
+	Cadence      string       `json:"cadence"`
+	Dependencies Dependencies `json:"dependencies"`
+	Arguments    Arguments    `json:"arguments"`
+}
+
+type FlowInteractionTemplate struct {
+	FType    string `json:"f_type"`
+	FVersion string `json:"f_version"`
+	ID       string `json:"id"`
+	Data     Data   `json:"data"`
+}
+
+func (t *FlowInteractionTemplate) IsScript() bool {
+	return t.Data.Type == "script"
+}
+
+func (t *FlowInteractionTemplate) IsTransaction() bool {
+	return t.Data.Type == "transaction"
+}
+
+func (t *FlowInteractionTemplate) GetAndReplaceCadenceImports(networkName string) (string, error) {
```

Comment:
https://github.com/onflow/flow-cli/blob/16b0c277df6c77d556e4c628dcdc50e373ba526e/flowkit/project/program.go#L74

Reply:
this is for executing the flix. Good catch we should have this "import" syntax in a helper lib that all tools import.

Reply:
Oh ok. Well then it might be ok. Since we kind of control how imports look like in a flix.